### PR TITLE
Add inline instructions to create item and auction it 

### LINKED ISSUE #57: INSTRUCTIONS: create item and auction it
Add inline instructions to create items, start an auction and bid on the item.

### PR DESCRIPTION
**Pull request by lakhina**
fixes #57

### CODE REVIEW

File: web-gateway/app/views/item.scala.html [hunk 1]
```diff
@@ -108,6 +109,7 @@
 
     @if(currentUser != item.getCreator && item.getStatus == ItemStatus.AUCTION) {
         <h3>@message("placeBid")</h3>
+<P>@message("instruction.placeBid")</P>
```
Comment (ignasi35) [inline web-gateway/app/views/item.scala.html]: Please review indentation.

File: web-gateway/app/views/createItem.scala.html [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
     @foundationForm(itemForm, routes.ItemController.createItem()) {
         <div class="column row">
+            <P>@message("instruction.createItem")</P>
```
Comment (ignasi35) [inline web-gateway/app/views/createItem.scala.html]: It'd be good to stick to a single tag capitalization approach. There's some `<P>` and other `<p>`. I think lowercase should be maintained.

Comment (lakhina) [inline web-gateway/app/views/createItem.scala.html]: thanks. I will take care of that

File: web-gateway/app/views/item.scala.html (hunk 1 above)
Comment (lakhina) [inline web-gateway/app/views/item.scala.html]: on it

File: web-gateway/conf/messages [hunk 3]
```diff
@@ -85,3 +85,6 @@ invalid.duration=Please enter a valid duration
 invalid.units=Please enter a valid unit
 
 createUserInstructions=Here you can create a new user who can participate in and create auctions.
+instruction.createItem=Here you can create the item which you want to auction.
+instruction.placeBid=Here you can bid a price for the item under auction.
+instruction.startAuction=Here you can start the auction of the item.
```
Comment (ignasi35) [inline web-gateway/conf/messages]: I think we can work this inline instructions to help the user understand better how starting an action and bidding works.
For example, once the auction has started it's not immediately clear that a user should switch accounts to proceed using the application.

also note that before the auction has started, the item page can be used for two purposes: starting the auction or editing the items details so the inline docs should reflect that.

Comment (lakhina) [inline web-gateway/conf/messages]: Thanks. I will try to get in the context and will update the PR accordingly

File: web-gateway/app/views/item.scala.html [hunk 4]
```diff
@@ -20,6 +20,7 @@
 
     <div >
         <h1>@item.getItemData.getTitle</h1>
+        <p>@message("instruction.startAuction")</p>
```
Comment (ignasi35) [inline web-gateway/app/views/item.scala.html]: I think this message is displayed unregarding the status of the auction.

Comment (ignasi35) [inline web-gateway/app/views/item.scala.html]: I created an item as user `bob` and started the auction. When I went to place a bid as `Alice` I could see this message.

screen_shot_2017-03-17_at_11_05_31

Comment (lakhina) [inline web-gateway/app/views/item.scala.html]: Thanks for pointing out, I will correct it.

**Comment by lakhina** [comment]
@ignasi35 Please review the changes and let me know if any changes required.

File: web-gateway/conf/messages [hunk 5]
```diff
@@ -87,3 +87,10 @@ invalid.units=Please enter a valid unit
 createUserInstructions=Here you can create a new user who can participate in and create auctions.
 itemMessage=Click on an item to view/edit its details.
 editItemInstructions=Here you can edit details of your auction item. If the auction has not started, you can edit any detail including description, minimum bid increment, auction duration, etc. However, if the auction is on-going, only the item description can be edited.
+instruction.createItem=Here you can create new items to auction.
+instruction.placeBid=Here you can place a bid for item under auction.
+instruction.seller=Here you can start the auction of the item.
+instruction.bidder.auctionStarted=The auction has started for this item.
+instruction.seller.auctionStarted=This item is under auction. Now other users can start bidding for this item.
+instruction.auctionEnded=The auction has ended for this item.
+instruction.auctionCancelled=The auction has been cancelled for this item.
```
Comment (ignasi35) [inline web-gateway/conf/messages]: I think the properties names are inconsistent with each other and with existing values. 
I like the `instructions.` prefix you introduced and I'd suggest we stick to it. I would redo the two following:
```
instruction.auctionStarted.bidder
instruction.auctionStarted.seller
```

Comment (lakhina) [inline web-gateway/conf/messages]: @ignasi35  I will update these.

instruction.auctionStarted.bidder
instruction.auctionStarted.seller

**Comment by lakhina** [comment]
@ignasi35  I have amended the changes. Please review

**Comment by ignasi35** [comment]
Thanks @lakhina.

Hi @mehulsr, while working on this PR @lakhina introduced a naming approach to `message`s keys that may be helpful going forward. We can consider it on other PRs related to instructions.

**Comment by mehulsr** [comment]
Hi @ignasi35 
Sure, will add it to the other PRs.
Thanks @lakhina  :smile:

**Comment by ignasi35** [comment]
@mehulsr well, some of the other PRs are already merged. We may want to adopt that pattern in future changes.

**Pull request merged**